src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
+            length: userQuery.length,
```

Comment:
Why store this separately rather than accessing it via `userQuery.length` when needed?

Reply:
It's a shorthand basically. Instead of writing `this.userQuery.length`, we simply write `this.length`.

Reply:
I moved it out of this object and renamed it `queryLength`.

Reply:
In this comment from January you said "I moved it out of this object and renamed it `queryLength`. But I don't see that change. As far as I can tell there is no type named `queryLength`.